src/fidesops/util/nested_utils.py

```diff
@@ -0,0 +1,40 @@
+from collections import defaultdict
+from typing import Dict, Any
+
+from fidesops.common_exceptions import FidesopsException
+
+
+def unflatten_dict(input_row: Dict[str, Any], separator: str = ".") -> Dict[str, Any]:
+    """
+    Takes a dictionary that has been normalized to level 1 and reconstitutes:
+
+    {
+       "a": 1,
+       "b.c": 2,
+       "b.d": 3
+    }
+
+    and turns it into:
+    {"a": 1, "b": {"c": 2, "d": 3}
+    """
+
+    def _create_dict() -> defaultdict:
+        """Can create a defaultdict at every level"""
+        return defaultdict(_create_dict)
+
+    unpacked_results = _create_dict()
+
+    for key, value in input_row.items():
+        levels = key.split(separator)
+        subdict = unpacked_results
```

Comment:
Woah this is a super elegant way of doing this!

Reply:
thank you!!